benchmarks/Microsoft.AspNetCore.Blazor.E2EPerformance/Pages/RenderList.cshtml

```diff
@@ -0,0 +1,75 @@
+@page "/renderlist"
+
+<h2>Render List</h2>
+
+Number of items: <input id="num-items" type="number" bind=@numItems />
+<button id="show-list" onclick=@Show>Show</button>
+<button id="hide-list" onclick=@Hide>Hide</button>
+
+@if (show)
+{
+    <table class='table'>
+        <thead>
+            <tr>
+                <th>Date</th>
+                <th>Temp. (C)</th>
+                <th>Temp. (F)</th>
+                <th>Summary</th>
+            </tr>
+        </thead>
+        <tbody>
+            @foreach (var forecast in GenerateForecasts(numItems))
+            {
+                <tr>
+                    <td>@forecast.DateFormatted</td>
+                    <td>@forecast.TemperatureC</td>
+                    <td>@forecast.TemperatureF</td>
+                    <td>@forecast.Summary</td>
+                </tr>
+            }
+        </tbody>
+    </table>
+}
+
+@functions {
+    int numItems = 10;
+    bool show = false;
+
+    void Hide()
+    {
+        show = false;
+    }
+
+    void Show()
+    {
+        show = true;
+    }
+
+    protected override void OnAfterRender()
+    {
+        BenchmarkEvent.Send("Finished rendering list");
+    }
+
+    static IEnumerable<WeatherForecast> GenerateForecasts(int count)
+    {
+        var rng = new Random();
```

Comment:
I'm not sure if its concerning that we're using the RNG as part of this benchmark. I don't think Random generally contributes too much perf overhead.

Reply:
Just to be sure, I've now removed the RNG from here.